src/cephadm/cephadm

```diff
@@ -2639,10 +2641,11 @@ def command_bootstrap():
                 f.write(ssh_pub)
             logger.info('Wrote public SSH key to to %s' % args.output_pub_ssh_key)
 
-            logger.info('Adding key to root@localhost\'s authorized_keys...')
-            if not os.path.exists('/root/.ssh'):
-                os.mkdir('/root/.ssh', 0o700)
-            auth_keys_file = '/root/.ssh/authorized_keys'
+            logger.info('Adding key to %s@localhost\'s authorized_keys...' % args.ssh_user)
+            user_home_dir = pathify('~%s' % (args.ssh_user))
+            if not os.path.exists(user_home_dir):
+                os.mkdir('%s/.ssh' % user_home_dir, 0o700)
+            auth_keys_file = '%s/.ssh/authorized_keys' % user_home_dir
```

Comment:
```suggestion
            ssh_dir = pathify('~%s/.ssh' % (args.ssh_user))
            if not os.path.exists(ssh_dir):
                os.makedirs(ssh_dir, 0o700)
            auth_keys_file = '%s/authorized_keys' % ssh_dir
```

Reply:
`/root` dir exists, but `/root/.ssh` dir does not exist

Reply:
i tested this section of code more thoroughly and found another issue.  if the user is running cephadm with sudo and /.ssh and authorized_keys dont exist and they are using the --ssh-user flag cephadm will create that dir and file with the root user as the owner which causes ssh connections to fail.

 I need to find a way to set the ownership of the dir/file to the user supplied with ssh-user flag